### PR TITLE
feat(components): add classname and style prop to all webex components

### PR DESCRIPTION
**Pull request by BenjaminGolya**
Added className and style prop to all webex components and place it on the outermost element (div) of each component.

https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-232579

### CODE REVIEW

File: src/components/WebexActivity/ActivityHeader.jsx [hunk 1]
```diff
@@ -43,15 +44,22 @@ export function formatMessageDate(timestamp) {
...
-export default function ActivityHeader({personID, timestamp}) {
+export default function ActivityHeader({className, personID, timestamp}) {
   const {displayName} = usePerson(personID);
 
+  const classBaseName = `${WEBEX_COMPONENTS_CLASS_PREFIX}-activity-header`;
+  const mainClasses = {
+    [classBaseName]: true,
+    [className]: !!className,
+  };
+
```
Comment (cipak) [inline src/components/WebexActivity/ActivityHeader.jsx:60]: I'd go with a simpler:
```js
const cssClasses = {
  [`${WEBEX_COMPONENTS_CLASS_PREFIX}-activity-header`]: true,
  [className]: !!className,
};
```

as I don't see anything "base" or main" about them.

Maybe we could also have a helper to cut down the boiler plate code:
```js
cssClasses = webexComponentClasses('activity-header', className, {...other-optional-classes...});
```

Let's see what @lalli-flores has to say about this.

File: src/components/helpers.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import classNames from 'classnames';
+
+import {WEBEX_COMPONENTS_CLASS_PREFIX} from '../constants';
+
+/**
+ * webex Component Classes defines class names for components
+ *
+ * @param {object} props Data passed to the component
```
Comment (cipak) [inline src/components/helpers.js:8]: double space

File: src/components/helpers.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * @param {string} props.classBaseName  Custom CSS class to apply
+ * @param {string} props.className  Custom style to apply
+ * @returns {object} Classes
+ */
+
+export default function webexComponentClasses(classBaseName, className, otherClasses) {
```
Comment (cipak) [inline src/components/helpers.js:14]: there are no props here, this function is not a component.

the signature should read:
```js
webexComponentClasses(componentBaseName, userClassName, otherClasses)
```

`otherClasses` is an object and must be listed in jsdoc.


I would advise building the complete class object like this:
```js
return classNames({
   [...]: true,
   [className]: !!className,
   ...otherClasses,
});

File: src/components/WebexInMeeting/WebexInMeeting.jsx [hunk 4]
```diff
@@ -13,24 +12,25 @@ import {TABLET, DESKTOP, DESKTOP_LARGE} from '../breakpoints';
...
-    [`${classBaseName}-tablet`]: width >= TABLET && width < DESKTOP,
-    [`${classBaseName}-desktop`]: width >= DESKTOP && width < DESKTOP_LARGE,
-    [`${classBaseName}-desktop-xl`]: width >= DESKTOP_LARGE,
+  const otherClasses = {
+    tablet: width >= TABLET && width < DESKTOP,
+    desktop: width >= DESKTOP && width < DESKTOP_LARGE,
+    'desktop-xl': width >= DESKTOP_LARGE,
     'remote-sharing': remoteShare !== null,
   };
+  const cssClasses = webexComponentClasses('in-meeting', className, otherClasses);
```
Comment (cipak) [inline src/components/WebexInMeeting/WebexInMeeting.jsx:30]: let's use a more compact format:
```js
const cssClasses = webexComponentClasses('in-meeting', className, {
  tablet: ...,
  desktop: ...,
  ...
});

File: src/components/WebexSettings/Select.jsx [hunk 5]
```diff
@@ -11,22 +10,27 @@ import {WEBEX_COMPONENTS_CLASS_PREFIX} from '../../constants';
...
  * @param {object} props  Data passed to the component
  * @param {string} props.className  Custom CSS class to apply
- * @param {Option[]} props.options  Array of options
- *
+ * @param {object[]} props.options  Array of options
+ * @param {string} props.type  Name of the control as defined in adapter
+ * @param {string} props.options[].value  The value of the option
+ * @param {string} props.options[].label  The label of the option
+ * @param {object} props.style  Custom style to apply
+>>>>>>> f12c480 (feat(components): add classname and style prop to all webex components)
```
Comment (cipak) [inline src/components/WebexSettings/Select.jsx:20]: this line does not look ok:)

File: src/components/WebexSettings/Tabs.jsx [hunk 6]
```diff
@@ -13,17 +13,30 @@ import {WEBEX_COMPONENTS_CLASS_PREFIX} from '../../constants';
...
  * @param {object} props  Data passed to the component
- * @param {Tab[]} props.tabs  The tabs to be rendered
- * @param {string} props.selected  Specifies which tab is active
+ * @param {string} props.className  Custom CSS class to apply
  * @param {Function} props.onSelect  A callback to be executed on selecting active tab
- *
+ * @param {string} props.selected  Specifies which tab is active
+ * @param {object} props.style  Custom style to apply
+ * @param {Tab[]} props.tabs  The tabs to be rendered
+ * @param {string} props.tabs[].key  Unique key for each tab
```
Comment (cipak) [inline src/components/WebexSettings/Tabs.jsx:21]: no need for these 2 lines, `Tab` is already declared above on line 7

File: src/components/WebexSettings/Tabs.jsx [hunk 7]
```diff
@@ -13,17 +13,30 @@ import {WEBEX_COMPONENTS_CLASS_PREFIX} from '../../constants';
...
+ * @param {string} props.tabs[].heading  Tab heading text
+>>>>>>> f12c480 (feat(components): add classname and style prop to all webex components)
```
Comment (cipak) [inline src/components/WebexSettings/Tabs.jsx:23]: check this

File: src/components/helpers.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import classNames from 'classnames';
+import {WEBEX_COMPONENTS_CLASS_PREFIX} from '../constants';
+
+/**
+ * webex Component Classes defines class names for components
+ *
+ * @param {object} props Data passed to the component
+ * @param {string} props.classBaseName Component class base name
```
Comment (cipak) [inline src/components/helpers.js:8]: double spaces for all params

File: src/components/helpers.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+ * @param {string} props.userClassName Custom CSS class to apply
+ * @param {string} props.otherClasses Other optional classes
+ * @returns {object} Classes
```
Comment (cipak) [inline src/components/helpers.js:11]: isn't it a string?

File: src/components/helpers.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import classNames from 'classnames';
+import {WEBEX_COMPONENTS_CLASS_PREFIX} from '../constants';
+
+/**
+ * webex Component Classes defines class names for components
```
Comment (cipak) [inline src/components/helpers.js:5]: Builds a string of css class names for webex components

File: src/components/helpers.js (hunk 9 above)
Comment (BenjaminGolya) [reply src/components/helpers.js:11]: classBaseName and userClassName is string, but otherClasses and the returned classes are objects

Comment (BenjaminGolya) [reply src/components/helpers.js:11]: yes it is

**Review by cipak — APPROVED** [review]

**Pull request merged**